charms/istio-pilot/src/charm.py

```diff
@@ -286,64 +294,126 @@ def handle_ingress_auth(self, event):
             )
             return
 
-        t = self.env.get_template('auth_filter.yaml.j2')
-        auth_filters = ''.join(
+        if self.model.config["ssl-crt"] and self.model.config["ssl-key"]:
+            gateway_port = GATEWAY_HTTPS_PORT
+        else:
+            gateway_port = GATEWAY_HTTP_PORT
+
+        t = self.env.get_template("auth_filter.yaml.j2")
+        auth_filters = "".join(
             t.render(
                 namespace=self.model.name,
                 app_name=self.app.name,
+                gateway_port=gateway_port,
                 **{
-                    'request_headers': yaml.safe_dump(
-                        [{'exact': h} for h in r.get('allowed-request-headers', [])],
+                    "request_headers": yaml.safe_dump(
+                        [{"exact": h} for h in r.get("allowed-request-headers", [])],
                         default_flow_style=True,
                     ),
-                    'response_headers': yaml.safe_dump(
-                        [{'exact': h} for h in r.get('allowed-response-headers', [])],
+                    "response_headers": yaml.safe_dump(
+                        [{"exact": h} for h in r.get("allowed-response-headers", [])],
                         default_flow_style=True,
                     ),
-                    'port': r['port'],
-                    'service': r['service'],
+                    "port": r["port"],
+                    "service": r["service"],
                 },
             )
             for r in auth_routes
         )
 
         self._resource_handler.delete_existing_resources(
             self._resource_handler.get_custom_resource_class_from_filename(
-                filename='auth_filter.yaml.j2'
+                filename="auth_filter.yaml.j2"
             ),
             namespace=self.model.name,
         )
         self._resource_handler.apply_manifest(auth_filters, namespace=self.model.name)
 
     @property
-    def _gateway_address(self):
-        """Look up the load balancer address for the ingress gateway.
-        If the gateway isn't available or doesn't have a load balancer address yet,
-        returns None.
-        """
-        # FIXME: service name is hardcoded
+    def _istiod_svc(self):
+        try:
+            exporter_service = self.lightkube_client.get(
+                res=Service, name="istiod", namespace=self.model.name
+            )
+            exporter_ip = exporter_service.spec.clusterIP
+        except ApiError as e:
+            if e.status.code == 404:
+                return None
+            raise
+        else:
+            return exporter_ip
```

Comment:
This was also introduced by another commit, I believe this comes from the observability integration.

Reply:
Removed